developomp/react-todo/src/globalContext.tsx

```diff
@@ -0,0 +1,93 @@
+import type { Dispatch, ReactNode, ReactElement } from "react"
+
+import { createContext, useReducer } from "react"
+import { nanoid } from "nanoid"
+
+/**
+ * Type definitions
+ */
+
+export interface TodoList {
+	[key: string]: string
+}
+
+export interface IGlobalState {
+	todo: TodoList
+}
+
+export type GlobalAction =
+	| {
+			type: ActionsEnum.ADD_TODO
+	  }
+	| {
+			type: ActionsEnum.REMOVE_TODO
+			payload: string
+	  }
+	| {
+			type: ActionsEnum.UPDATE_TODO
+			payload: { key: string; content: string }
+	  }
+
+export interface IGlobalContext {
+	globalState: IGlobalState
+	dispatch: Dispatch<GlobalAction>
+}
+
+export enum ActionsEnum {
+	ADD_TODO,
+	REMOVE_TODO,
+	UPDATE_TODO,
+}
+
+export enum LocalStorageKey {
+	TODO = "todo",
+}
+
+/**
+ * Code starts here
+ */
+
+function saveToLocalStorage(todo: any) {
+	window.localStorage.setItem(LocalStorageKey.TODO, JSON.stringify(todo))
+}
+
+const defaultState: IGlobalState = {
+	todo: JSON.parse(window.localStorage.getItem(LocalStorageKey.TODO) as any),
```

Comment:
```suggestion
	todo: JSON.parse(window.localStorage.getItem(LocalStorageKey.TODO) as any) || [],
```
여기서 빈배열을 주도록 하면 될것 같네요